File: summer/WebContent/js/system/windows/controls/primitives/IContainItemStorage.js
/**
 * IContainItemStorage
 */

define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	var IContainItemStorage = declare("IContainItemStorage", null,{
//		void 
		StoreItemValue:function(/*object*/ item, /*DependencyProperty*/ dp, /*object*/ value){},
//		object 
		ReadItemValue:function(/*object*/ item, /*DependencyProperty*/ dp){},
//		void 
		ClearItemValue:function(/*object*/ item, /*DependencyProperty*/ dp){},
//		void 
		ClearValue:function(/*DependencyProperty*/ dp){},
//		void 
		Clear:function(){},
	});
	
	IContainItemStorage.Type = new Type("IContainItemStorage", IContainItemStorage, [Object.Type], true);
	return IContainItemStorage;
});

File: org.summer.view.widget/src/org/summer/view/widget/controls/primitives/IContainItemStorage.java
package org.summer.view.widget.controls.primitives;

import org.summer.view.widget.DependencyProperty;

public interface IContainItemStorage {
	void Clear();

	void ClearItemValue(Object item, DependencyProperty dp);

	void ClearValue(DependencyProperty dp);

	Object ReadItemValue(Object item, DependencyProperty dp);

	void StoreItemValue(Object item, DependencyProperty dp, Object value);
}
